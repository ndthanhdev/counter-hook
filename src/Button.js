import React from 'react'
import PropTypes from 'prop-types'

function Button({ className, text, onClick = () => {} }) {
  return (
    <button className={className} onClick={onClick}>
      {text}
    </button>
  )
}

Button.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func,
}

export default Button
