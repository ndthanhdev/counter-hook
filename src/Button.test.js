import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Button from './Button'

test('render', () => {
  const tree = renderer.create(<Button text="test"w/>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('should invoke onClick when click on', () => {
  const mockHandleClick = jest.fn()
  const wrapper = shallow(<Button text="test" onClick={mockHandleClick} />)
  wrapper.find('button').simulate('click')
  expect(mockHandleClick).toHaveBeenCalledTimes(1)
})
