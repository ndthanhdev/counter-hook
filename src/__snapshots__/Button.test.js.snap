// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<button
  onClick={[Function]}
>
  test
</button>
`;
